llvm/lib/Transforms/InstCombine/InstCombineCasts.cpp

```diff
@@ -862,6 +862,20 @@ Instruction *InstCombiner::visitTrunc(TruncInst &Trunc) {
   if (Instruction *I = foldVecTruncToExtElt(Trunc, *this))
     return I;
 
+  // FIXME: This is temporary work-around for a problem reported here:
+  // https://github.com/KhronosGroup/SPIRV-LLVM-Translator/issues/645
+  //
+  // InstCombine canonical form for this pattern
+  // ```
+  //   // Example (little endian):
+  //   //   trunc (extractelement <4 x i64> %X, 0) to i32
+  //   //   --->
+  //   //   extractelement <8 x i32> (bitcast <4 x i64> %X to <8 x i32>), i32 0
+  // ```
+  // can't be lowered by SPIR-V translator to "standard" format.
+  if (Trunc.getModule()->getTargetTriple().substr(0, 4) == "spir")
```

Comment:
Nit: you can use `StringRef(...).startswith("spir")`. A little bit more readable.